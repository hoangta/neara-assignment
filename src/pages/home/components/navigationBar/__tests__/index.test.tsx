import React from "react";
import { render, fireEvent } from '@testing-library/react';
import NavigationBar from '..'

describe('Pages/Home/Components/NavigationBar', () => {
    const onEditSpy = jest.fn();
    const onCreateSpy = jest.fn();

    it('should render correctly', () => {
        const { asFragment } = render(<NavigationBar onEdit={onEditSpy} onCreate={onCreateSpy} />);
        expect(asFragment()).toMatchSnapshot();
    })

    it('should trigger onEdit when click Edit button', () => {
        const { getByTestId } = render(<NavigationBar onEdit={onEditSpy} onCreate={onCreateSpy} />);
        const editButton = getByTestId('edit_button');
        fireEvent.click(editButton);
        expect(onEditSpy).toHaveBeenCalledTimes(1);
    })

    it('should trigger onCreate when click Add button', () => {
        const { getByTestId } = render(<NavigationBar onEdit={onEditSpy} onCreate={onCreateSpy} />);
        const addButton = getByTestId('add_button');
        fireEvent.click(addButton);
        expect(onCreateSpy).toHaveBeenCalledTimes(1);
    })
})
