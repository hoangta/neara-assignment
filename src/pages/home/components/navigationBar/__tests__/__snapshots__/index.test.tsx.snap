// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pages/Home/Components/NavigationBar should render correctly 1`] = `
<DocumentFragment>
  <div
    class="container"
  >
    <button
      class="button"
      data-testid="edit_button"
    >
      Edit
    </button>
    <div
      class="title"
    >
      Tokens
    </div>
    <button
      class="button"
      data-testid="add_button"
      style="font-size: 25px;"
    >
      ï¼‹
    </button>
  </div>
</DocumentFragment>
`;
